scrape/manager.go

```diff
@@ -140,7 +141,7 @@ func (m *Manager) UnregisterMetrics() {
 
 func (m *Manager) reloader() {
 	reloadIntervalDuration := m.opts.DiscoveryReloadInterval
-	if reloadIntervalDuration < model.Duration(5*time.Second) {
+	if !m.opts.skipDiscoveryReloadIntervalCheck && reloadIntervalDuration < model.Duration(5*time.Second) {
```

Comment:
instead of a flag, I'm suggesting to remove this hidden min value https://github.com/prometheus/prometheus/pull/13147/files#r1404481390